### PR TITLE
Removes cudf dependency for CPU Ops

### PR DESCRIPTION
**Pull request by albert17**

### CODE REVIEW

**Comment by albert17** [comment]
run tests

File: nvtabular/dispatch.py [hunk 1]
```diff
@@ -17,16 +17,20 @@
...
 import pyarrow.parquet as pq
-from cudf.core.column import as_column, build_column
-from cudf.utils.dtypes import is_list_dtype
+
+try:
+    import cudf
+    from cudf.core.column import as_column, build_column
+    from cudf.utils.dtypes import is_list_dtype
+except ImportError:
+    cudf = None
```
Comment (benfred) [inline nvtabular/dispatch.py:33]: We'll also need to add support for cudf being None to all functions that use it in this file - and assume we're CPU only if not set.

File: nvtabular/dispatch.py [hunk 2]
```diff
@@ -366,3 +376,23 @@ def _to_host(x):
...
+
+def _from_host(x):
+    if isinstance(x, cudf.DataFrame):
+        return x
+    else:
+        return cudf.DataFrame.from_arrow(x)
+
+
+def _get_list_dtype():
+    return cudf.core.dtypes.ListDtype
```
Comment (benfred) [inline nvtabular/dispatch.py:389]: We should add a pandas equivalent here.

**Comment by albert17** [comment]
@benfred updated with changes

**Comment by albert17** [comment]
run tests

**Review by benfred — APPROVED** [review]

**Pull request merged**